File: packages/http-client-csharp-generator/test/CadlRanchMockApis/src/PetStore-TypeSpec.ts
import { passOnSuccess, ScenarioMockApi, mockapi, json, MockApi } from "@azure-tools/cadl-ranch-api";

export const Scenarios: Record<string, ScenarioMockApi> = {};

const dog = {
  name: "dog",
  age: 12,
};

Scenarios.PetStore_DeletePetById = passOnSuccess(
  mockapi.delete("/pets/1", () => {
    return {
      status: 200,
    };
  }),
);

Scenarios.PetStore_ReadPetById = passOnSuccess(
  mockapi.get("/pets/1", () => {
    return {
      status: 200,
      body: json(dog),
    };
  }),
);

Scenarios.PetStore_CreatePet = passOnSuccess(
  mockapi.post("/pets", (req) => {
    req.expect.bodyEquals(dog);
    return {
      status: 200,
      body: json(dog),
    };
  }),
);

Scenarios.PetStore_GetPetByKind = passOnSuccess(
  mockapi.get("/pets/dog", () => {
    return {
      status: 200,
      body: json(dog),
    };
  }),
);

Scenarios.PetStore_GetFirstPet = passOnSuccess(
  mockapi.get("/pets", (req) => {
    req.expect.containsQueryParam("start", "1");
    return {
      status: 200,
      body: json(dog),
    };
  }),
);

Scenarios.PetStore_GetFish = passOnSuccess(
  mockapi.get("/pets/getFish", (req) => {
    req.expect.containsQueryParam("kind", "shark");
    return {
      status: 200,
      body: json({
        kind: "shark",
        size: 100,
        bite: "I can bite"
      }),
    };
  }),
);

File: packages/http-client-csharp-generator/test/CadlRanchProjects.Tests/PetStore-TypeSpec.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading.Tasks;
using AutoRest.TestServer.Tests.Infrastructure;
using PetStore;
using NUnit.Framework;
using Azure;
using PetStore.Models;

namespace CadlRanchProjects.Tests
{
    public class PetStoreTypeSpecTests : CadlRanchMockApiTestBase
    {
        [Test]
        public Task PetStore_DeletePetById() => Test(async (host) =>
        {
            Response response = await new PetStoreClient(host, null).GetPetsClient().DeleteAsync(1);
            Assert.AreEqual(200, response.Status);
        });

        [Test]
        public Task PetStore_ReadPetById() => Test(async (host) =>
        {
            Response response = await new PetStoreClient(host, null).GetPetsClient().ReadAsync(1, new RequestContext());
            Assert.AreEqual(200, response.Status);
            Assert.AreEqual(12, Pet.FromResponse(response).Age);
            Assert.AreEqual("dog", Pet.FromResponse(response).Name);
        });

        [Test]
        public Task PetStore_CreatePet() => Test(async (host) =>
        {
            Pet pet = new("dog", 12);
            var response = await new PetStoreClient(host, null).GetPetsClient().CreateAsync(pet);
            Assert.AreEqual("dog", response.Value.Name);
            Assert.AreEqual(12, response.Value.Age);
        });

        [Test]
        public Task PetStore_GetPetByKind() => Test(async (host) =>
        {
            Pet pet = new("dog", 12);
            Response response = await new PetStoreClient(host, null).GetPetsClient().GetPetByKindAsync("dog");
            Assert.AreEqual(200, response.Status);
            Assert.AreEqual(12, Pet.FromResponse(response).Age);
            Assert.AreEqual("dog", Pet.FromResponse(response).Name);
        });

        [Test]
        public Task PetStore_GetFirstPet() => Test(async (host) =>
        {
            Pet pet = new("dog", 12);
            Response response = await new PetStoreClient(host, null).GetPetsClient().GetFirstPetAsync(1, new RequestContext());
            Assert.AreEqual(200, response.Status);
            Assert.AreEqual(12, Pet.FromResponse(response).Age);
            Assert.AreEqual("dog", Pet.FromResponse(response).Name);
        });

        [Test]
        public Task PetStore_GetFish() => Test(async (host) =>
        {
            Response<Fish> response = await new PetStoreClient(host, null).GetPetsClient().GetFishAsync("shark");
            Assert.AreEqual(200, response.GetRawResponse().Status);
            Assert.IsTrue(response.Value is Shark);
            Shark shark = response.Value as Shark;
            Assert.AreEqual(100, shark.Size);
            Assert.AreEqual("I can bite", shark.Bite);
        });
    }
}
